10-actors-in-a-distributed-system/projects/final/Sky/ScanModel.swift

```diff
@@ -60,62 +58,76 @@ class ScanModel: ObservableObject {
   /// Completed scan tasks.
   @MainActor @Published var completed = 0
 
+  @MainActor @Published var localTasksCompleted: [String] = []
+
   @Published var total: Int
 
-  // MARK: - Methods
+  let actorSystem: BonjourActorSystem
 
   init(total: Int, localName: String) {
     self.total = total
-    let localSystem = ScanSystem(name: localName)
-    systems = Systems(localSystem)
-    service = ScanTransport(localSystem: localSystem)
-    service.taskModel = self
+    self.actorSystem = BonjourActorSystem(localName: localName)
     systemConnectivityHandler()
   }
 
-  func run(_ task: ScanTask) async throws -> String {
-    Task { @MainActor in scheduled += 1 }
-    defer {
-      Task { @MainActor in scheduled -= 1 }
-    }
-    return try await systems.localSystem.run(task)
-  }
-
   func systemConnectivityHandler() {
     Task {
-      for await notification in
-        NotificationCenter.default.notifications(named: .connected) {
-        guard let name = notification.object as? String else { continue }
-        print("[Notification] Connected: \(name)")
-        await systems.addSystem(name: name, service: self.service)
+      for await count in actorSystem.$actorCount.values {
         Task { @MainActor in
-          isConnected = await systems.systems.count > 1
+          isConnected = count > 1
         }
       }
     }
 
     Task {
-      for await notification in
-        NotificationCenter.default.notifications(named: .disconnected) {
-        guard let name = notification.object as? String else { return }
-        print("[Notification] Disconnected: \(name)")
-        await systems.removeSystem(name: name)
+      for await notification in NotificationCenter.default
+        .notifications(named: .localTaskUpdate) {
+        let status = notification.taskStatus
+        let runningTasksCount = try await actorSystem.localActor.count
         Task { @MainActor in
-          isConnected = await systems.systems.count > 1
+          if scheduled == 0 {
+            isCollaborating = runningTasksCount > 0
+          }
+          localTasksCompleted.append(status)
         }
       }
     }
   }
 
+  func worker(number: Int, actor: ScanActor) async
+  -> Result<Data, ScanTaskError> {
+    await onScheduled()
+
+    let task = ScanTask(input: number)
+
+    let result: Result<Data, ScanTaskError>
+    do {
+      result = try .success(await actor.run(task))
+    } catch {
+      result = .failure(runAllTasks.init(
```

Comment:
@freak4pc functions don't have inits - this is definitely a typo?

Reply:
Seems like it. Very strange, wondering how it even compiled for me.